modules/engine/src/lib/animation-loop.js

```diff
@@ -126,55 +127,48 @@ export default class AnimationLoop {
     return this;
   }
 
+  start(opts = {}) {
+    this._start(opts);
+    return this;
+  }
+
   // Starts a render loop if not already running
   // @param {Object} context - contains frame specific info (E.g. tick, width, height, etc)
-  start(opts = {}) {
+  async _start(opts) {
     if (this._running) {
       return this;
     }
     this._running = true;
     // console.debug(`Starting ${this.constructor.name}`);
     // Wait for start promise before rendering frame
-    const startPromise = this._getPageLoadPromise()
-      .then(() => {
-        if (!this._running || this._initialized) {
-          return null;
-        }
-
-        // Create the WebGL context
-        this._createWebGLContext(opts);
-        this._createFramebuffer();
-        this._startEventHandling();
-
-        // Initialize the callback data
-        this._initializeCallbackData();
-        this._updateCallbackData();
-
-        // Default viewport setup, in case onInitialize wants to render
-        this._resizeCanvasDrawingBuffer();
-        this._resizeViewport();
-
-        this._gpuTimeQuery = Query.isSupported(this.gl, ['timers']) ? new Query(this.gl) : null;
-
-        this._initialized = true;
-
-        // Note: onIntialize can return a promise (in case it needs to load resources)
-        return this.onInitialize(this.animationProps);
-      })
-      .then((appContext) => {
-        if (this._running) {
-          this._addCallbackData(appContext || {});
-          if (appContext !== false) {
-            this._startLoop();
-          }
-        }
-      });
+    try {
+      await this._getPageLoadPromise();
 
-    if (this.props.onError) {
-      startPromise.catch(this.props.onError);
-    }
+      this._initialize(opts);
 
-    return this;
+      // Note: onIntialize can return a promise (in case it needs to load resources)
+      const appContext = await this.onInitialize(this.animationProps);
+
+      this._addCallbackData(appContext || {});
+
+      // check that we haven't been stopped
+      if (!this._running) {
+        return null;
+      }
+
+      // Start the loop
+      if (appContext !== false) {
+        // cancel any pending renders to ensure only one loop can ever run
+        this._cancelAnimationFrame();
+        this._requestAnimationFrame();
+      }
+
+      return this;
+    } catch (error) {
+      this.props.onError(error);
```

Comment:
Is it a breaking change here ( throw -> console.error )?

Reply:
No, the intention was to keep the semantics unaffected. As far as I can tell, the previous code had a `promise.catch` statement that neutralized the exception (i.e. the rejected promise).